src/content/docs/controller.md

```diff
@@ -2,4 +2,218 @@
 title: Controller
 category: 02. Hardware
 page: 11
----
\ No newline at end of file
+---
+
+Controllers are the basis of how you control the V5 Brain remotely. You can upload programs and communicate with the competition switch using the controller. The controller has a screen, buttons, and joysticks that you can program to interact with the V5 Brain and its peripherals. The controller is also used to control the robot during the driver control period (opcontrol).
+
+During a match, one or two controllers can be used to control the robot. We'll refer to these controllers as the _primary controller_ and the _partner controller_. Having a partner controller is optional, but it can be useful for controlling additional functions of your robot. These two controllers are connected together with a smart cable.
+
+# Getting a reference to a `Controller`
+
+We already talked about the [peripherals API](/docs/peripherals) earlier as providing a safe way to interact with the V5 Brain's peripherals. Using it, you can access the controller as follows:
+
+```rs
+// @fold start
+#![no_std]
+#![no_main]
+
+use vexide::prelude::*;
+
+// @fold end
+#[vexide::main]
+async fn main(peripherals: Peripherals) {
+    //                           (                )
+    let controller = peripherals.primary_controller;
+    //                               ^
+    //         [Get the primary controller. Note that since we moved the controller out of `peripherals`, we can no longer access it there.]
+    //                                   (                )
+    let partner_controller = peripherals.partner_controller;
+}
+```
+
+# Reading the controller's state
+
+<!-- TODO: add one of those fancy illustrations that look cool here. -->
+
+Reading the controller's state is simple through the `Controller::state` method. This method returns a `ControllerState` struct that contains the state of the controller's buttons and joysticks.
+
+You can read each button's state through the `button_x` properties and the joysticks' state through the `y_stick` properties on the `ControllerState` struct.
+
+Here's an example of how you can read the controller's state in a loop:
+
+```rs
+// @fold start
+#![no_std]
+#![no_main]
+
+use vexide::prelude::*;
+// @fold end
+#[vexide::main]
+async fn main(peripherals: Peripherals) {
+  let mut controller = peripherals.primary_controller;
+
+  loop {
+    //          (                )
+    let state = controller.state();
```

Comment:
I think this code is broken because you don't handle the result.

Reply:
Writing code without linting is a pain!